### PR TITLE
Add Speech Commands Notebook

### PR DESCRIPTION
**Pull request by titu1994**
# Changelog
- All ASR models now properly dispatch and serialize the updated vocabulary / labels.
  - Without this, upon `restore_from` / `load_from_checkpoint` after a modification to the vocabulary, the model instantly crashes.
  - This is because the weights are for the new vocabulary but config is for old vocabulary.
  - Model config therefor instantiated is for old vocabulary, and incompatible with weight shape during deserialization
- Load from checkpoint is now wrapped by the same restoration block - so as to disable construction of the data loaders during restoration from a PTL `load_from_checkpoint`.
  - Note, if signatures change in the future, we need to update this code signature too. Otherwise no additional maintenance is necessary
- setup_optimization() will now preserve the passed config (if not None), assuming it to be an updated version. 
  - This is essential so that user does not need to override the internal ModelPT._cfg.optim by hand.
  - The notebook does show both methods, but the first method will be preferred.
- For requirements, transformers becomes a general requirement so that both ASR and NLP can depend on it.
  - Enables use of nemo_toolkit[asr]

# Notebook
The notebook is now a thorough example of multiple things - 

1) Demonstration of setting up training from scratch
2) Saving and restoring best practices using experiment manager / `save_to` and `restore_from`.
3) Demonstration of fine tuning on new data by preserving encoder.

Signed-off-by: smajumdar <titu1994@gmail.com>

### CODE REVIEW

File: tutorials/asr/02_Speech_Commands.ipynb [hunk 1]
```diff
@@ -0,0 +1,1069 @@
...
+        "\"\"\"\n",
+        "# If you're using Google Colab and not running locally, run this cell.\n",
+        "\n",
+        "## Install dependencies\n",
+        "!pip install wget\n",
+        "!apt-get install sox libsndfile1 ffmpeg\n",
+        "!pip install unidecode\n",
+        "\n",
+        "# ## Install NeMo\n",
+        "!python -m pip install --upgrade git+https://github.com/NVIDIA/NeMo.git@candidate#egg=nemo_toolkit[all]\n",
```
Comment (blisc) [inline tutorials/asr/02_Speech_Commands.ipynb:43]: Why not nemo_toolkit[asr]?

**Review by percy21081993 — APPROVED** [review]
**

**Review by percy21081993 — COMMENTED** [review]
**

File: tutorials/asr/02_Speech_Commands.ipynb (hunk 1 above)
Comment (titu1994) [reply tutorials/asr/02_Speech_Commands.ipynb:43]: Ahh I just used the ASR tutorial header and removed the hash, good point.

Comment (titu1994) [reply tutorials/asr/02_Speech_Commands.ipynb:43]: Seemingly `tokenizers` is not part of nemo_toolkit[asr]. 

```python
---------------------------------------------------------------------------
ModuleNotFoundError                       Traceback (most recent call last)
<ipython-input-8-327743e4f6ae> in <module>()
      3 # NeMo's ASR collection - this collections contains complete ASR models and
      4 # building blocks (modules) for ASR
----> 5 import nemo.collections.asr as nemo_asr

6 frames
/usr/local/lib/python3.6/dist-packages/nemo/collections/common/tokenizers/tokenizer_utils.py in <module>()
     16 from typing import List, Optional
     17 
---> 18 from transformers import AlbertTokenizer, BertTokenizer, DistilBertTokenizerFast, RobertaTokenizer
     19 
     20 import nemo

ModuleNotFoundError: No module named 'transformers'

---------------------------------------------------------------------------
NOTE: If your import is failing due to a missing package, you can
manually install dependencies using either !pip or !apt.

To view examples of installing some common dependencies, click the
"Open Examples" button below.
---------------------------------------------------------------------------
```

**Comment by titu1994** [comment]
@blisc The readme has some major changes, please re-review when time permits.

**Pull request merged**